hummingbird/ml/convert.py

```diff
@@ -164,33 +155,77 @@ def _convert_onnxml(model, backend, test_input, device, extra_config={}):
     # Generate some test input if necessary.
     if test_input is None:
         if backend == onnx.__name__:
-            assert (
-                initial_types is not None and not initial_types[0][1].shape is None
-            ), "Cannot generate test input data. Initial_types do not contain shape information."
-            assert len(initial_types[0][1].shape) == 2, "Hummingbird currently support only inputs with len(shape) == 2."
-
             from onnxconverter_common.data_types import FloatTensorType, DoubleTensorType, Int32TensorType, Int64TensorType
 
-            test_input = np.random.rand(initial_types[0][1].shape[0], initial_types[0][1].shape[1])
+            if initial_types is None:
+                # Generate inputs from the ONNX schema.
+                initial_types = []
+                for input in model.graph.input:
+                    name = input.name if hasattr(input, "name") else None
+                    data_type = (
+                        input.type.tensor_type.elem_type
+                        if hasattr(input, "type")
+                        and hasattr(input.type, "tensor_type")
+                        and hasattr(input.type.tensor_type, "elem_type")
+                        else None
+                    )
+                    if name is None:
+                        raise RuntimeError(
+                            "Cannot fetch input name or data_type from the ONNX schema. Please provide some test input."
```

Comment:
If we remove support for `ONNX_INITIAL_TYPES`, this won't be possible right?

Reply:
No no, this will be the default. So we can remove the `if initial_types is None` condition and just generated the types directly from the model